### PR TITLE
Cache subfolders

### PR DESCRIPTION
**Pull request by MartinNemi03**
Added type subfolders for /cache folder with some folder optimization in mind.
This is not applied in development mode, so it won't spam you with folders. 
It's mainly meant for the main site, where we can have tens of thousands of files in a single folder.

### CODE REVIEW

File: src/helper.js [hunk 1]
```diff
@@ -983,6 +985,40 @@ export function parseItemTypeFromLore(lore) {
   };
 }
 
+export function getCacheFilePath(dirPath, type, name) {
+  // we don't care about folder optimalization when we're developing
```
Comment (dukio) [inline src/helper.js]: optimization typo

File: src/helper.js [hunk 2]
```diff
@@ -983,6 +985,40 @@ export function parseItemTypeFromLore(lore) {
...
+  if (process.env?.NODE_ENV == "development") {
+    return path.resolve(dirPath, `${type}_${name}.png`);
+  }
+
+  let subdirs = [type];
```
Comment (dukio) [inline src/helper.js]: const?

File: src/helper.js [hunk 3]
```diff
@@ -983,6 +985,40 @@ export function parseItemTypeFromLore(lore) {
...
+  if (type == "leather" || type == "potion") {
+    subdirs.push(name.split("_")[0]);
+  }
+
+  // check if the entire folder path is available
+  if (!fs.pathExistsSync(path.resolve(dirPath, subdirs.join("/")))) {
+    // check if every subdirectory is available
+    for (let i = 1; i <= subdirs.length; i++) {
+      let checkDirs = subdirs.slice(0, i);
+      let checkPath = path.resolve(dirPath, checkDirs.join("/"));
```
Comment (dukio) [inline src/helper.js]: all const?

**Review by dukio — CHANGES_REQUESTED** [review]
I'm on mobile but I noticed these things... will check/test further when on desktop

File: src/helper.js (hunk 2 above)
Comment (MartinNemi03) [reply src/helper.js]: not sure, since i potentially push into it, but i'll try.

File: src/helper.js (hunk 1 above)
Comment (MartinNemi03) [reply src/helper.js]: ty, my english broken.

File: src/helper.js (hunk 3 above)
Comment (MartinNemi03) [reply src/helper.js]: yes, all const.

File: src/helper.js (hunk 2 above)
Comment (dukio) [reply src/helper.js]: Yeah pushing is not an issue, only thing you can't do is reassign the value completely

**Review by dukio — APPROVED** [review]
didn't test but looks good

**Review by metalcupcake5 — APPROVED** [review]

**Pull request merged**